File: myjava/src/main/java/com/oldboy/java/domain/Dog.java
package com.oldboy.java.domain;

/**
 * Created by Administrator on 2018/9/5.
 */
public class Dog  extends Animal{
	public Dog(String name){
		super(name);
		System.out.println("new Dog() : " + name);
	}
	public Dog(){
		this("tom");
	}
	public void run() {
		System.out.println("~~~~");
	}
}

File: myscala/src/main/java/Dog.scala
import scala.beans.BeanProperty

/**
  * Created by Administrator on 2018/9/29.
  */
class Dog {

    private var age = 0
    private var name = "tom"
    var color = "white"

    //生成标准get/set方法
    @BeanProperty
    var blood = "A"

    def increment = {
        age +=1
    }
    def current() = age

    def getAge() = {
        age
    }
    def setAge(age:Int) = {
        this.age =age
    }


}
